src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -92,8 +93,11 @@ const AppFrame = ({ appUrl, allowedFeaturesList }: AppFrameProps): ReactElement
       message: string | EIP712TypedData,
       requestId: string,
       method: Methods.signMessage | Methods.signTypedMessage,
+      sdkVersion: string,
     ) => {
-      openSignMessageModal(message, requestId, method, !!settings.offChainSigning)
+      const isOffChainSigningSupported = isOffchainEIP1271Supported(safe, chain, sdkVersion)
+      const signOffChain = isOffChainSigningSupported && !useOnChainSigning
+      openSignMessageModal(message, requestId, method, signOffChain && !!settings.offChainSigning)
```

Comment:
```suggestion
      openSignMessageModal(message, requestId, method, signOffChain)
```

Do we still need to keep the state variable https://github.com/safe-global/safe-wallet-web/blob/56c74af38ef6a33306a0e19a31752979ccc81105/src/components/safe-apps/AppFrame/index.tsx#L60-L63 ?

Reply:
These are needed. The `settings` (preferences) are [set by the Safe App](https://docs.safe.global/learn/safe-core/safe-core-protocol/signatures/eip-1271#enabling-off-chain-signing).